src/TEdit/Editor/Plugins/RandomizerPlugin.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using TEdit.Terraria;
+using TEdit.ViewModel;
+
+
+namespace TEdit.Editor.Plugins
+{
+    public class RandomizerPlugin : BasePlugin
+    {
+        public RandomizerPlugin(WorldViewModel worldViewModel)
+            : base(worldViewModel)
+        {
+            Name = "Randomize All Blocks in the World";
+        }
+
+        public override void Execute()
+        {
+            if (_wvm.CurrentWorld == null)
+                return;
+
+            RandomizerPluginView view = new();
+            if (view.ShowDialog() == false)
+                return;
+
+            RandomizerSettings settings = new()
+            {
+                Seed = view.Seed
+            };
+
+            var mapping = GetRandomBlockMapping(settings);
+
+            for (int x = 0; x < _wvm.CurrentWorld.Size.X; x++)
```

Comment:
I would suggest using the selection area if one is set, so that the user can specify a region to randomize.